components/JointPetitionForm.vue

```diff
@@ -0,0 +1,474 @@
+    buttonText: {
+      type: String,
+      required: false,
+      default: function () {
+        return this.$t('form.button_cta')
+      }
+    },
+    hasComment: {
+      type: Boolean,
+      required: false,
+      default: false
+    },
+    hasCompany: {
+      type: Boolean,
+      required: false,
+      default: false
+    },
+    hasCompanyToggle: {
+      type: Boolean,
+      required: false,
+      default: false
+    },
+    tweetText: {
+      type: String,
+      required: false,
+      default: null
+    },
+    letterText: {
+      type: String,
+      required: false,
+      default: function () {
+        return this.$t('global.letter_text')
+      }
+    }
+  },
+
+  data() {
+    return {
+      isSending: false,
+      hasSigned: false,
+      errorMessage: null,
+      comment: null,
+      isBusinessOwner: false,
+      companyName: null,
+      referrerGroup: null
+    }
+  },
+
+  computed: {
+    ...mapState(['donateUrl']),
+
+    routeName() { return this.$nuxt.$route.name },
+
+    name: {
+      get() {
+        return this.$store.state.name
+      },
+      set(value) {
+        this.$store.commit('setName', value)
+      }
+    },
+
+    email: {
+      get() {
+        return this.$store.state.email
+      },
+      set(value) {
+        this.$store.commit('setEmail', value)
+      }
+    },
+
+    address: {
+      get() {
+        return this.$store.state.address
+      },
+      set(value) {
+        this.$store.commit('setAddress', value)
+      }
+    },
+
+    zipCode: {
+      get() {
+        return this.$store.state.zipCode
+      },
+      set(value) {
+        this.$store.commit('setZipCode', value)
+      }
+    },
+
+    phone: {
+      get() {
+        return this.$store.state.phone
+      },
+      set(value) {
+        this.$store.commit('setPhone', value)
+      }
+    },
+
+    shouldContactCongress() {
+      return this.contactCongress.toLowerCase() === 'yes' ? 1 : 0
+    },
+
+    mothershipTags() {
+      return JSON.stringify(Object.values(this.tags))
+    },
+
+    referrer() {
+      return this.$route.query.referrer
+    },
+
+    petitionString() {
+      return this.$t('petition_url_string')
+    },
+
+    firstName() {
+      const words = (this.name || '').split(' ')
+      return words.shift()
+    },
+
+    lastName() {
+      const words = (this.name || '').split(' ')
+      words.shift()
+      return words.join(' ')
+    }
+  },
+
+  watch: {
+    name(newValue) {
+      setValue('form-first_name', this.firstName)
+      setValue('form-last_name', this.lastName)
+    },
+
+    email(newValue) {
+      setValue('form-email', newValue)
+    },
+
+    address(newValue) {
+      setValue('form-street', newValue)
+    },
+
+    zipCode(newValue) {
+      setValue('form-zip_code', newValue)
+    },
+
+    comment(newValue) {
+      setValue('form-comments', newValue)
+    }
+  },
+
+  created() {
+    this.comment = this.letterText
+    this.hasSigned = this.hasSignedAlready
```

Comment:
I think this is leftover from when I was testing something. Should be safe to remove.